libraries/azurerm_virtual_network.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'azurerm_resource'
+
+class AzurermVirtualNetwork < AzurermSingularResource
+  name 'azurerm_virtual_network'
+  desc 'Verifies settings for an Azure Virtual Network'
+  example <<-EXAMPLE
+    describe azurerm_virtual_network(resource_group: 'example', name: 'vnet-name') do
+      it { should have_monitoring_agent_installed }
+    end
+  EXAMPLE
+
+  ATTRS = %i(
+    id
+    name
+    location
+    type
+    tags
+    properties
+  ).freeze
+
+  attr_reader(*ATTRS)
+
+  def initialize(resource_group: nil, name: nil)
+ 
+    resp = client.virtual_network(resource_group, name)
+    return if resp.nil? || resp.key?('error')
+
+    ATTRS.each do |field|
+      instance_variable_set("@#{field}", resp[field.to_s])
+    end
+    
+    @subs = begin
+             resp['properties']['subnets']
+            rescue
+                 []
+            end
+    @exists = true
+  end
+
+  def address_space
+    properties['addressSpace']['addressPrefixes']
+  end
+
+  def dns_servers
+      properties['dhcpOptions']['dnsServers']
+  end
+      
+  def vnet_peerings
+    properties['virtualNetworkPeerings'].each_with_object({}) do |peer, hash|
+        hash[ peer['name'] ] = peer['properties']['remoteVirtualNetwork']['id']
+    end
```

Comment:
Props for using `each_with_object` rather than creating an empty accumulator outside the loop!

In this library we prefer to use Ruby's functional capabilities, like this:
```ruby
def vnet_peerings
  name_id = ->(peer) { [peer['name'], peer.dig('properties', 'remoteVirtualNetwork', 'id')] }

  properties['virtualNetworkPeerings'].collect(&name_id).to_h
end
```
We use `collect` to make an array of whatever comes back while iterating over the given array. The lambda, `pair`, returns an array of the `name`/`id` pair, and when those are collected we turn the result into a hash.

Separately, you can see I also changed it to use `dig`, which helps us avoid `NoMethodError` in the event one of the intermediary keys doesn't exist—sadly, this is something the Azure APIs do—but you may want to introduce some filtering to ensure there aren't any nil entries that have to be worked around by consumers of this method.

Something like:
```ruby
def vnet_peerings
  name_id = ->(peer) { [peer['name'], peer.dig('properties', 'remoteVirtualNetwork', 'id')] }
  any_nils = ->(pair) { pair.any?(&:nil) }

  properties['virtualNetworkPeerings'].collect(&name_id).reject(&any_nils).to_h
end
```